nrf52840-hal/src/ieee802154.rs

```diff
@@ -38,11 +35,23 @@ pub const DEFAULT_TXPOWER: TxPower = TxPower::_0dBm;
 /// Default Start of Frame Delimiter = `0xA7` (IEEE compliant)
 pub const DEFAULT_SFD: u8 = 0xA7;
 
+/// note: picked trough trial & error
+pub const DEFAULT_ED_THRESHOLD: u8 = 0x50;
+
+/// Default Clear Channel Assessment method = Carrier sense
+pub const DEFAULT_CCA: Cca = Cca::CarrierSense;
+
 // TODO expose the other variants in `pac::CCAMODE_A`
 /// Clear Channel Assessment method
 pub enum Cca {
     /// Carrier sense
     CarrierSense,
+    /// Energy above the Energy Detection threshold. Energy measurements above this value mean
+    /// that the channel is assumed to be busy.
+    /// Note the the measurement range is 0..0xFF - where 0 means that the received power was
+    /// less than 10 dB above the selected receiver sensitivity. This does not correlate to dBm,
+    /// see the nrf52840 Product Specification for conversion instructions.
+    EnergyDetection { ed_threshold: u8 },
```

Comment:
you can and should also document the field of the variant. That is:

``` rust
/// This is an enum
enum Enum {
    /// This is one of the variants
    Variant {
        /// This is the field of the variant
        field: u8,
    }
}
```
Check out the `cargo doc`. These three doc comments render differently.